### PR TITLE
[JAVA][FEIGN]Implement unit tests for java-feign client

### PR DESCRIPTION
**Pull request by hugo-ma-alves**
Hello

This PR includes the following changes

- Unit tests for java feign client
- Implement the unit tests with Junit5
- Set default feign http client to OkHttp, the default feign client does not support Patch verb

cc:

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @nmuesch (2021/01)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Java/libraries/feign/api_test.mustache [hunk 1]
```diff
@@ -32,7 +33,7 @@ public class {{classname}}Test {
      * {{notes}}
      */
     @Test
-    public void {{operationId}}Test() {
+    void {{operationId}}Test() {
```
Comment (borsch) [inline modules/openapi-generator/src/main/resources/Java/libraries/feign/api_test.mustache:36]: class also can be package-private

Comment (hugo-ma-alves) [reply modules/openapi-generator/src/main/resources/Java/libraries/feign/api_test.mustache:36]: Thanks,
I will make that change.

File: samples/client/petstore/java/feign/src/test/java/org/openapitools/client/api/AnotherFakeApiTest.java [hunk 2]
```diff
@@ -1,9 +1,10 @@
 package org.openapitools.client.api;
+//TODO update packages
```
Comment (wing328) [inline samples/client/petstore/java/feign/src/test/java/org/openapitools/client/api/AnotherFakeApiTest.java:2]: @hugo-ma-alves thanks for the PR, which looks good overall.

I saw the TODO here. What needs to be done?

Comment (hugo-ma-alves) [reply samples/client/petstore/java/feign/src/test/java/org/openapitools/client/api/AnotherFakeApiTest.java]: It was a reminder to remove the unused imports. All done now

File: modules/openapi-generator/src/main/resources/Java/libraries/feign/api_test.mustache [hunk 3]
```diff
@@ -1,10 +1,11 @@
 package {{package}};
+//TODO update packages
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/feign/api_test.mustache:2]: @hugo-ma-alves last question from me. Should this line be removed from the template?

Comment (hugo-ma-alves) [reply modules/openapi-generator/src/main/resources/Java/libraries/feign/api_test.mustache]: Yes, removed

**Comment by wing328** [comment]
CircleCI failure not related to this change.

**Comment by wing328** [comment]
Notice that the indentation has changed from 4-space to 2-space in the test files. Users are advised to use a code formatter (e.g. prettier.io) to automatically format the auto-generated code to make the code conforms to code style set by the users.

**Pull request merged**